File: 0001_0100/0024/0024.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    n, m = (int(x) for x in f.read().split())
    q.write(str(1 if m == 0 else n if m == 1 else n // (m - 1) * (2 * n - m + 1 + n // (m - 1) * (1 - m)) // 2))

File: 0001_0100/0024/0024.go
package main

import (
	"fmt"
	"os"
)

func main() {
	f, _ := os.Open("input.txt")
	q, _ := os.Create("output.txt")
	var n, m int
	fmt.Fscan(f, &n, &m)
	if m == 0 {
		n = 1
	} else if m > 1 {
		n = n / (m - 1) * (2*n - m + 1 + n/(m-1)*(1-m)) / 2
	}
	fmt.Fprint(q, n)
}
